### ISSUE TITLE
Issue witth query

### ISSUE THREAD

**Issue by afidegnum** [joowani/python-arango#49]
HI @joowani can you please give a hand on the following aql query? i asked the arangodb community for the past many hours and got no reply yet, 

based on the previous question i posted, i built a product taxonomy graph, but trying to traverse the edges, brings an error. can you please give a hint?

```
for t in inbound taxonomy catof
return t
```

`[1577], 'collection 'taxonomy' used as expression operand'`
`[10], 'Invalid input for traversal: Only id strings or objects with _id are allowed'`

**Comment by joowani** [comment]
Hi @afidegnum,

Could you post your your code snippet please? I will need more context.

Thanks,
Joohwan

**Comment by afidegnum** [comment]
ok, let me explain more in details, 

here is an example of a [taxonomy list ](https://www.google.com/basepages/producttype/taxonomy.en-US.txt)based, on the questions i asked few days ago. 

here is my graph structure
edge `catof` is a hirarchical edge which fetch children from parents. (`taxonomy`)
{"_from":"taxonomy/5254567","_to":"taxonomy/5257887" 
{"_from":"taxonomy/5246912","_to":"taxonomy/5256129"}
{"_from":"taxonomy/5251175","_to":"taxonomy/5245634"}
{"_from":"taxonomy/5255451","_to":"taxonomy/5256990"}

and product catalogs `catalogs` with edge  `belong_to`
{"_from":"catalogs/5377977","_to":"taxonomy/5254882"} 
{"_from":"catalogs/5352582","_to":"taxonomy/5254882"}
{"_from":"catalogs/5347646","_to":"taxonomy/5254882"}

using python arango, i was stuck in the middle, so i resorted in aql, which also ended up in error. 

what i need to do is fetch the parent taxonomy who do not have edge relations. and traverse taxonomy parent to the corresponding product. 

that's the situation i m facing.

**Comment by joowani** [comment]
Have you tried `Graph.traverse`? What was the problem you were having with python-arango before you tried AQL? As for the AQL query itself I'm not an expert in the field so you are better off asking the ArangoDB peeps.

**Comment by afidegnum** [comment]
yes. i had a look at the traverse feature of python-arango with this example 

```
# Traverse the graph in outbound direction, breath-first
traversal_results = schedule.traverse(
    start_vertex='profs/michelle',
    direction='outbound',
    strategy='bfs',
    edge_uniqueness='global',
    vertex_uniqueness='global',
)
print(traversal_results)
```

but what of traversing from documents collections linked by edges?

**Comment by joowani** [comment]
Sorry I don't understand your question re: but what of traversing from documents collections linked by edges? Could you rephrase it? 

Also regarding this: 

```
what i need to do is fetch the parent taxonomy who do not have edge relations. and traverse taxonomy parent to the corresponding product.
```

Could you also give me a concrete example of what result you expect from what input (using a category from the CSV for example)? Thanks.

**Comment by afidegnum** [comment]
hi @joowani i have posted the question on[ Stack Overflow,](https://stackoverflow.com/questions/45585738/taxonomy-hierarchical-vertices-how-do-i-fetch-top-parent-and-selective-childre) can you please assist there?

**Comment by joowani** [comment]
Hi @afidegnum, ok I will check it out. Closing this one out.
